### PR TITLE
[Dotnet] XamarinShot port

### PR DESCRIPTION
**Pull request by stephen-hawley**
Yes, 531 files. Sorry, you gotta dance with them what brung ya

### CODE REVIEW

File: dotnet/ios/ios12/XamarinShot/XamarinShot/Utils/SimdExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,227 @@
+﻿namespace XamarinShot.Utils;
+
+public static class SimdExtensions
+{
+        public static SCNMatrix4 SetTranslation (this SCNMatrix4 matrix, OpenTK.Vector3 value)
+        {
+                // only for column-major matrixes
```
Comment (rolfbjarne) [inline dotnet/ios/ios12/XamarinShot/XamarinShot/Utils/SimdExtensions.cs]: Looks like this sample will have to be updated after the next preview (to pick up our SCNMatrix4 changes).

File: dotnet/ios/ios12/XamarinShot/XamarinShot/Main.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    using UIKit;
+
+    public class Application
+    {
+        // This is the main entry point of the application.
+        static void Main(string[] args)
+        {
+            // if you want to use a different Application Delegate class from "AppDelegate"
+            // you can specify it here.
+            UIApplication.Main(args, null, "AppDelegate");
```
Comment (rolfbjarne) [inline dotnet/ios/ios12/XamarinShot/XamarinShot/Main.cs:13]:
```suggestion
            UIApplication.Main(args, null, typeof (AppDelegate));
```

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: dotnet/ios/ios12/XamarinShot/XamarinShot/XamarinShot.csproj [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+	<BundleResource Include="Resources\Sounds\slingshot_swish_05.wav" />
+	<BundleResource Include="Resources\Sounds\vortex_04.m4a" />
+	<BundleResource Include="Resources\Launch Screens\iPad.png" />
+	<BundleResource Include="Resources\Launch Screens\iPhone_Landscape.png" />
+	<BundleResource Include="Resources\Launch Screens\iPhone_Portrait.png" />
+	<BundleResource Include="Resources\Launch Screens\iPhone_X_Landscape.png" />
+	<BundleResource Include="Resources\Sounds\music_menu.m4a" />
+  </ItemGroup>
+  <ItemGroup>
+	<SceneKitAsset Include="art.scnassets\blocks\entities\block_archA.scn" />
```
[the remark this answers is not in the document]
Comment (stephen-hawley) [reply dotnet/ios/ios12/XamarinShot/XamarinShot/XamarinShot.csproj]: Yes, yes I have. Without this, it doesn't run.

File: dotnet/ios/ios12/XamarinShot/XamarinShot/Core/Local Networking/NetworkSession.cs [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+                }
+        }
+
+        #endregion
+
+        #region IMCNearbyServiceAdvertiserDelegate
+
+        public void DidReceiveInvitationFromPeer (MCNearbyServiceAdvertiser advertiser, MCPeerID peerID, NSData? context, MCNearbyServiceAdvertiserInvitationHandler invitationHandler)
+        {
+                invitationHandler (true, Session);
```
Comment (tj-devel709) [inline dotnet/ios/ios12/XamarinShot/XamarinShot/Core/Local Networking/NetworkSession.cs:210]: Maybe this method should be capitalized

File: dotnet/ios/ios12/XamarinShot/XamarinShot/Core/Local Networking/ProximityManager.cs [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+public interface IProximityManagerDelegate
+{
+        void LocationChanged (ProximityManager manager, GameTableLocation location);
+
+        void AuthorizationChanged (ProximityManager manager, bool authorization);
+}
+
+public class ProximityManager : CLLocationManagerDelegate
+{
+        readonly NSUuid RegionUUID = new NSUuid ("53FA6CD3-DFE4-493C-8795-56E71D2DAEAF");
```
Comment (tj-devel709) [inline dotnet/ios/ios12/XamarinShot/XamarinShot/Core/Local Networking/ProximityManager.cs:12]:
```suggestion
        readonly NSUuid RegionUuid = new NSUuid ("53FA6CD3-DFE4-493C-8795-56E71D2DAEAF");
```

File: dotnet/ios/ios12/XamarinShot/XamarinShot/Core/World Map Loading/KeyPositionAnchor.cs [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        public KeyPositionAnchor (UIImage image, OpenTK.NMatrix4 transform, ARWorldMappingStatus mappingStatus) : base ("KeyPosition", transform)
+        {
+                Image = image;
+                MappingStatus = mappingStatus;
+        }
+
+        // this is guaranteed to be called with something of the same class
+        public KeyPositionAnchor (ARAnchor anchor) : base (anchor)
+        {
+                var other = anchor as KeyPositionAnchor;
```
Comment (tj-devel709) [inline dotnet/ios/ios12/XamarinShot/XamarinShot/Core/World Map Loading/KeyPositionAnchor.cs:14]: good is operator use

**Review by tj-devel709 — COMMENTED** [review]
I heard that the average American scrolls the height of the Statue of Liberty every day...
After this review, I feel like I have to at least be at 3 Statue of Liberty's ahah

**Comment by rolfbjarne** [comment]
[Microsoft support for Xamarin](https://dotnet.microsoft.com/en-us/platform/support/policy/xamarin) will end on May 1, 2024 for all Xamarin SDKs. In preparation for this, all issues and PRs in this repository are being closed.

**Pull request closed without merging**